src/eui/Eui64.cc

```diff
@@ -75,22 +80,99 @@ Eui::Eui64::lookupSlaac(const Ip::Address &c)
 
 // return binary representation of the EUI
 bool
-Eui::Eui64::lookupNdp(const Ip::Address &/*c*/)
+Eui::Eui64::lookupNdp(const Ip::Address &c)
 {
-#if 0 /* no actual lookup coded yet */
-
-    /* no OS yet supported for NDP protocol lookup */
+    bool success = false;
+
+#if _SQUID_LINUX_
+    int rtnetlink_sock;
+    struct {
+        struct nlmsghdr hdr;
+        struct ndmsg ndm;
+    } req;
+    struct sockaddr_nl sa;
+    struct iovec iov;
+    struct msghdr msg;
+    char buf[16384];
```

Comment:
Is there any particular reason for using a 16K buffer here on the stack?
Please consider using SBuf or MemBuf for large-ish buffers that cannot be "static char". That will also allow some later adjustments to allocate based on read() needs rather than magic numbers.